File: Seminar/src/main/java/org/sopt/Seminar/global/auth/filter/SoptExceptionHandler.java
package org.sopt.Seminar.global.auth.filter;

import static jakarta.servlet.http.HttpServletResponse.*;

import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.io.IOException;
import org.springframework.security.access.AccessDeniedException;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.security.web.access.AccessDeniedHandler;
import org.springframework.stereotype.Component;

@Component
public class SoptExceptionHandler implements AccessDeniedHandler, AuthenticationEntryPoint {
    @Override
    public void commence(HttpServletRequest request, HttpServletResponse response,
                         AuthenticationException authException) throws IOException, ServletException {
        setResponse(response, SC_UNAUTHORIZED);
    }

    @Override
    public void handle(HttpServletRequest request, HttpServletResponse response,
                       AccessDeniedException accessDeniedException) throws IOException, ServletException {
        setResponse(response, SC_FORBIDDEN);
    }

    private void setResponse(HttpServletResponse response, int code) {
        response.setStatus(code);
    }

}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/global/auth/filter/SoptExceptionHandler.kt
package org.sopt.kotlinsemina.global.auth.filter

import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.access.AccessDeniedException
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.AuthenticationEntryPoint
import org.springframework.security.web.access.AccessDeniedHandler
import org.springframework.stereotype.Component

@Component
class SoptExceptionHandler: AuthenticationEntryPoint, AccessDeniedHandler {
    override fun commence(
        request: HttpServletRequest?,
        response: HttpServletResponse?,
        authException: AuthenticationException?
    ) {
        setResponse(response!!, HttpServletResponse.SC_UNAUTHORIZED)
    }

    override fun handle(
        request: HttpServletRequest?,
        response: HttpServletResponse?,
        accessDeniedException: AccessDeniedException?
    ) {
        setResponse(response!!, HttpServletResponse.SC_FORBIDDEN)
    }

    private fun setResponse(response: HttpServletResponse, code: Int) {
        response.status = code
    }
}